File: Interview_bit/Array/Pascal_triangle.cpp
#include <bits/stdc++.h>
using namespace std;

// Pseudo Code
// For Row i, Row[i][0] = Row[i][i] = 1. 
// And Row[i][j] = Row[i-1][j] + Row[i-1][j-1], when j belongs to [1, i)


int main(){
    vector<vector<int>> X;
    for(int i=1; i<=5;i++){
        vector<int> B;
        for(int j=0; j<i; j++){
            if(i == 1 || i==2){
                B.push_back(1);
            }
            else{
                if(j==0 || j==i-1){
                    B.push_back(1);
                }else{
                    B.push_back(X[i-2][j-1] + X[i-2][j]);
                }
            }    
        }
        X.push_back(B);
    }
    for(int i=0; i<X.size(); i++){
        for(int j=0; j<X[i].size(); j++){
            cout << X[i][j] << " ";
        }
        cout<<"\n";
    }
    return 0;
}

File: Interview_bit/Array/Pascal_triangle.py
x = int(input())
A = []

# Pseudo Code
# For Row i, Row[i][0] = Row[i][i] = 1. 
# And Row[i][j] = Row[i-1][j] + Row[i-1][j-1], when j belongs to [1, i)

for i in range(1,x+1):
    B = []
    for j in range(i):
        if(i == 1 or i==2):
            B.append(1)
        else:
            if (j==0 or j ==i-1):
                B.append(1)
            else:
                B.append(A[i-2][j-1] + A[i-2][j])
    A.append(B)
for i in A:
    print(i)